cli/src/test/all-tests.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+#set -eu
+set -o pipefail
+
+source "$(dirname "${0}")"/functions-test.sh
+
+MATRIX_OS="$1"
+# Switch IDEasy binary file name based on github workflow matrix.os name (first argument of all-tests.sh)
+BINARY_FILE_NAME="ideasy"
+if [ "${MATRIX_OS}" == "windows-latest" ]; then
+  BINARY_FILE_NAME="ideasy.exe"
+fi
+
+START_TIME=$(date '+%Y-%m-%d_%H-%M-%S')
+
+DEBUG_INTEGRATION_TEST_PREFIX="${HOME}/tmp/ideasy-integration-test-debug"
+DEBUG_INTEGRATION_TEST="${DEBUG_INTEGRATION_TEST_PREFIX}-${START_TIME}"
+IDE_HOME="${DEBUG_INTEGRATION_TEST}/home-dir"
+export IDE_ROOT="${IDE_HOME}/projects"
+IDEASY_DIR="${IDE_ROOT}/_ide"
+FUNCTIONS="${IDEASY_DIR}/installation/functions"
+IDE="${DEBUG_INTEGRATION_TEST}/home-dir/projects/_ide/bin/${BINARY_FILE_NAME}"
+TEST_RESULTS_FILE="${IDE_ROOT}/testResults"
+
+test_files_directory=$(realpath "$0" | xargs dirname)
+
+success=0
+failure=0
+total=0
+
+function doTestsInner() {
+  # Note: requires var test_files_directory to be set.
+  echo "Entering doTestsInner..."
+  for testpath in "${test_files_directory:?}/integration-tests"/*; do
+    testcase="${testpath/*\//}"
+    echo "Running test #${total}: ${testcase} (${testpath})"
+    # Every test runs in its own environment.
+    # Need to return back from created environment (integration test might have changed pwd).
+    #echo "Switching directory to: ${IDE_ROOT}"
+    #cd "${IDE_ROOT:?}" || exit
+
+    integration_test_result=0
+
+    # Following line adds a trap to ERR signal: Whenever error (some
+    # conditions apply; see documentation) set integration_test_result=1.
+    trap 'doWarning "A non-handled error in integration test occurred."; integration_test_result=1' ERR
+
+    source "${testpath:?}"
+
+    # Remove trap
+    # Disable errexit in case it was turned on in a test script
+    trap - ERR
+    set +e
+
+    echo "RESULTS:"
+    echo "$integration_test_result"
```

Comment:
Should this `echo` be visible in the logs or is it there for debugging?

Reply:
I've removed it now.